presto-hive/src/main/java/com/facebook/presto/hive/CreateEmptyPartitionProcedure.java

```diff
@@ -0,0 +1,123 @@
+import java.util.Optional;
+
+import static com.facebook.presto.spi.StandardErrorCode.ALREADY_EXISTS;
+import static com.facebook.presto.spi.block.MethodHandleUtil.methodHandle;
+import static com.facebook.presto.spi.type.StandardTypes.VARCHAR;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public class CreateEmptyPartitionProcedure
+        implements Provider<Procedure>
+{
+    private static final MethodHandle INSERT_EMPTY_PARTITION = methodHandle(
+            CreateEmptyPartitionProcedure.class,
+            "createEmptyPartition",
+            ConnectorSession.class,
+            String.class,
+            String.class,
+            List.class,
+            List.class);
+
+    private final HiveMetadataFactory hiveMetadataFactory;
+    private final ExtendedHiveMetastore metastore;
+    private final LocationService locationService;
+    private final JsonCodec<PartitionUpdate> partitionUpdateJsonCodec;
+
+    @Inject
+    public CreateEmptyPartitionProcedure(HiveMetadataFactory hiveMetadataFactory, ExtendedHiveMetastore metastore, LocationService locationService, JsonCodec<PartitionUpdate> partitionUpdateCodec)
+    {
+        this.hiveMetadataFactory = requireNonNull(hiveMetadataFactory, "hiveMetadataFactory is null");
+        this.metastore = requireNonNull(metastore, "metastore is null");
+        this.locationService = requireNonNull(locationService, "locationService is null");
+        this.partitionUpdateJsonCodec = requireNonNull(partitionUpdateCodec, "partitionUpdateCodec is null");
+    }
+
+    @Override
+    public Procedure get()
+    {
+        return new Procedure(
+                "system",
+                "create_empty_partition",
+                ImmutableList.of(
+                        new Argument("schema", VARCHAR),
+                        new Argument("table", VARCHAR),
+                        new Argument("partitionColumnNames", "array(varchar)"),
+                        new Argument("partitionValues", "array(varchar)")),
+                INSERT_EMPTY_PARTITION.bindTo(this));
+    }
+
+    public void createEmptyPartition(ConnectorSession session, String schema, String table, List<Object> partitionColumnNames, List<Object> partitionValues)
+    {
+        HiveMetadata hiveMetadata = hiveMetadataFactory.create();
+
+        HiveInsertTableHandle hiveInsertTableHandle = hiveMetadata.beginInsert(session, new HiveTableHandle(schema, table));
+
+        List<String> actualPartitionColumnNames = hiveInsertTableHandle.getInputColumns().stream()
+                .filter(HiveColumnHandle::isPartitionKey)
+                .map(HiveColumnHandle::getName)
+                .collect(toImmutableList());
+        checkArgument(Objects.equals(partitionColumnNames, actualPartitionColumnNames), "input partition column names doesn't match actual partition column names");
```

Comment:
I think this needs to be a `PrestoException(INVALID_PROCEDURE_ARGUMENT)` as it means they are passing in invalid partition names. Add a test for this.